packages/fether-react/src/Send/TxForm/TxForm.js

```diff
@@ -115,18 +170,27 @@ class Send extends Component {
                           <Field
                             centerText={`${values.gasPrice} GWEI`}
                             className='-range'
-                            label='Transaction Fee'
-                            leftText='Slow'
+                            label='Transaction Speed'
+                            leftText='Low'
                             max={MAX_GAS_PRICE}
                             min={MIN_GAS_PRICE}
                             name='gasPrice'
                             render={FetherForm.Slider}
                             required
-                            rightText='Fast'
+                            rightText='High'
                             step={0.5}
                             type='range' // In Gwei
                           />
 
+                          <OnChange name='gasPrice'>
+                            {(value, previous) => {
+                              if (this.state.maxSelected) {
+                                const args = { toggleMax: false };
```

Comment:
To be clearer, I would:
- create a method called `toggleMax`, which will only setState
- in button onClick, call `this.toggleMax();  mutators.recalculateMax(); // without args`
- remove setState from `mutators.recalculateMax()`, recalculateMax should only change the `amount` field
- in OnChange, `mutators.recalculateMax(); // without args`

Reply:
Understood, I guess it's simply not clean to have different logic in the same function.